src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -270,6 +270,9 @@ public function get($id, $invalidBehavior = self::EXCEPTION_ON_INVALID_REFERENCE
 
                     $alternatives = array();
                     foreach ($this->services as $key => $associatedService) {
+                        if (isset($this->privates[$key])) {
```

Comment:
> Yes. Why did you changed ?

Waiting for upstream to change :stuck_out_tongue:, but ill add it. Makes sense.

Got your point about `getServiceIds`. It's public API.. which should keep returning privates till 4.0. Ill have a look at it.